<?xml version="1.0"?>

<robot name="dobot" xmlns:xacro="http://www.ros.org/wiki/xacro">
<xacro:property name="yaml_file" value="parameters.yaml"/>
<xacro:property name="params" value="${load_yaml(yaml_file)}"/>
  <link name="base_link">
      <visual>
        <geometry>
	   <box size="${params.base_link_size}"/>
	</geometry>
      </visual>
  </link>
  <link name="platform">
      <visual>
        <geometry>
	  <box size="${params.platform_size}"/>
	</geometry>
	<origin xyz="${params.platform_origins}"/>
      </visual>
  </link>
  <link name="arm1">
      <visual>
        <geometry>
	   <cylinder length="${params.arm1_size[0]}" radius="${params.arm1_size[1]}"/>
	</geometry>
	<origin xyz="${params.arm1_origins}"/>
      </visual>
  </link>
  <link name="arm2">
      <visual>
        <geometry>
	  <cylinder length="${params.arm2_size[0]}" radius="${params.arm2_size[1]}"/> 
	</geometry>
	<origin xyz="${params.arm2_origins}"/>
      </visual>
  </link>
  <link name="fourth">
    <visual>
      <geometry>
        <box size="${params.fourth_size}"/>
      </geometry>
      <origin xyz="${params.fourth_origins}"/>
    </visual>
  </link>
  <link name="gripper">
      <visual>
        <geometry>
	  <!-- <cylinder length="${params.gripper_size[0]}" radius="${params.gripper_size[1]}"/> -->
    <box size="${params.gripper_size}"/>
	</geometry>
	<origin xyz="${params.gripper_origins[0]}" rpy="${params.gripper_origins[1]}"/>
      </visual>
  </link>
  
  <joint name="joint1" type="revolute">
   <parent link="base_link" />
   <child link="platform" />
	<limit effort="${params.effort}" lower="${params.joint1_limits[1]}" upper="${params.joint1_limits[0]}" 	velocity="${params.joint1_velocity[0]}"/>
	<axis xyz="${params.joint1_axis}"/>
	<origin xyz="${params.joint1_origins}"/> 
  </joint>
  
  <joint name="joint2" type="revolute">
   <parent link="platform" />
   <child link="arm1" />
	<limit effort="${params.effort}" lower="${params.joint2_limits[0]}" upper="${params.joint2_limits[1]}" velocity="${params.joint2_velocity[0]}"/>
	<axis xyz="${params.joint2_axis}"/>
	<origin xyz="${params.joint2_origins}"/>
  </joint>

  <joint name="joint3" type="revolute">
   <parent link="arm1" />
   <child link="arm2" />
	<limit effort="${params.effort}" lower="${params.joint3_limits[1]}" upper="${params.joint3_limits[0]}" velocity="${params.joint3_velocity[0]}"/>
	<axis xyz="${params.joint3_axis}"/>
	<origin xyz="${params.joint3_origins}"/>
  </joint>

  <joint name="joint5" type="revolute">
   <parent link="arm2" />
   <child link="fourth" />
  <limit effort="${params.effort}" lower="${params.joint5_limits[0]}" upper="${params.joint5_limits[1]}" velocity="${params.joint3_velocity[0]}"/>
  <axis xyz="${params.joint5_axis}"/>
  <origin xyz="${params.joint5_origins}"/>
  </joint>

  <joint name="joint4" type="revolute">
   <parent link="fourth" />
   <child link="gripper" />
	<limit effort="${params.effort}" lower="${params.joint4_limits[1]}" upper="${params.joint4_limits[0]}" velocity="${params.joint4_velocity[0]}"/>
	<axis xyz="${params.joint4_axis}"/>
	<origin xyz="${params.joint4_origins}"/>
  </joint>
</robot>
